File: VolleyballAPI/VolleyballAPI/Entities/PlayerDetails.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using VolleyballAPI.Entities;

namespace VolleyballManagementAppBackend.Entities
{
  public class PlayerDetails
  {
    [Required]
    [Key]
    public Guid Id { get; set; }

    [Required]
    [ForeignKey(nameof(User))]
    public Guid UserId { get; set; }
    public DateTime Birthday { get; set; }
    public string Phone { get; set; }
    public int PlayerNumber { get; set; }
    public TicketPass TicketPass { get; set; }
    public Gender Gender { get; set; }
    public ICollection<Post> Posts { get; set; }

    public virtual User User { get; set; }
    public virtual ICollection<TeamPlayer> Teams { get; set; }

    }
}

File: VolleyballManagementAppFrontend/src/app/models/player-details.ts
import { Gender } from "./gender";
import { Post } from "./post";
import { TicketPass } from "./ticket-pass";

export interface PlayerDetails {
    id: string;
    userId: string;
    birthday: Date;
    phone: string;
    playerNumber: number;
    ticketPass?: TicketPass;
    gender: Gender;
    posts: Post[];
}